### ISSUE TITLE
Does not work with xelatex

### ISSUE THREAD

**Issue by tetrapharmakon** [paolobrasolin/tex-ipenama#3]
At the moment there is no way to make ipenama compile with engines different than pdftex. Maybe something like
```ipenama [-latex|-xelatex|-context] file.tex```
would be of help?

**Comment by paolobrasolin** [comment]
I confirm the issue, but the problem resides not within `ipenama`.

As you might know `latexmk` is a `perl` script.
Try to locate it and open it; near the top you'll find 
```
## Commands to invoke latex, pdflatex, etc
$latex  = 'latex %O %S';
$pdflatex = 'pdflatex %O %S';
$lualatex = 'lualatex %O %S';
# xelatex is used to give xdv file, not pdf file
$xelatex = 'xelatex -no-pdf %O %S';
```

These are `perl` strings containing the _templates_ for the commandlines that `latexmk` runs when selecting the respective engine. The `%S` is where the _filename_ goes.

To run `ipenama` you need to run `findhyph` on the compilation log, but in order for that to work you must compile your document using a magic flag:
```
latexmk -r "ipenama.latexmkrc" main.tex
```

That `-r` flag tells `latexmk` to read a configuration file named `ipenama.latexmkrc`. What is a `latexmk` configuration file? Another perl script, _of course_. 🙄

Let's take a look at the relevant lines of `ipenama.latexmkrc`:
```
$latex    =~ s/%S/"\\tracingparagraphs=1\\input{%S}"/;
$pdflatex =~ s/%S/"\\tracingparagraphs=1\\input{%S}"/;
```
I'm applying a regexp (`=~`) to the string containing the commandline templates: `%S` is replaced by `"\tracingparagraphs=1\input{%S}"`. That is, instead of compiling the document by passing its basename (`%S`) we compile a `TeX` string literally (that's what the double quotes are for). This string is a minimal `TeX` document that does two things:
```
\tracingparagraphs=1 % enable p. tracing output in the log files (read by findhyph)
\input{%S}           % input your document's main file and carry on
```
Basically, that's a thin wrapper that enables paragraph tracing and nothing else.
Note I'm using a regexp instead of simply changing the values to minimize interference with other `latexmkrc`s that might be loading in your setup.

I think you're now seeing what the problem is: we're not automagically enabling paragraph tracing for the `XeLaTeX` engine since we're not patching the `$xelatex`.

I'm assigning you this issue.

**TODO:**

- [ ] fix `ipenama.latexmkrc` by adding the missing string replacements for `$lualatex` and `$xelatex`,
- [ ] check the mechanism is working correctly with both `LuaLaTeX` and `XeLaTeX`.

Please make a pull request from your fork when you're done and I'll review and merge it. Thanks. ❤️
